src/main/java/org/json/JSONWriter.java

```diff
@@ -328,8 +328,7 @@ public static String valueToString(Object value) throws JSONException {
             // Instead we will quote it as a string
             return JSONObject.quote(numberAsString);
         }
-        if (value instanceof Boolean || value instanceof JSONObject
-                || value instanceof JSONArray) {
+        if (isDirectlyConvertibleToJSON(value)) {
```

Comment:
Please revert this change. A separate method is not needed for this conditional. You can add a clarification comment if you want.